src/olympia/addons/tests/test_views.py

```diff
@@ -1236,6 +1236,61 @@ def test_filter_by_recommended(self):
         assert len(data['results']) == 1
         assert data['results'][0]['id'] == addon.pk
 
+    def test_filter_by_promoted(self):
+        av_min, _ = AppVersion.objects.get_or_create(
+            application=amo.ANDROID.id, version='59.0.0')
+        av_max, _ = AppVersion.objects.get_or_create(
+            application=amo.ANDROID.id, version='60.0.0')
+
+        addon = addon_factory(name='Recomménded Addôn')
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.ANDROID.id, version=addon.current_version,
+            min=av_min, max=av_max)
+        self.make_addon_promoted(addon, RECOMMENDED, approve_version=True)
+
+        addon2 = addon_factory(name='Fírefox Addôn')
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.ANDROID.id, version=addon2.current_version,
+            min=av_min, max=av_max)
+        self.make_addon_promoted(addon2, RECOMMENDED, approve_version=True)
+        addon2.promotedaddon.update(application_id=amo.FIREFOX.id)
+
+        addon3 = addon_factory(slug='other-addon', name=u'Other Addôn')
+        ApplicationsVersions.objects.get_or_create(
+            application=amo.ANDROID.id, version=addon3.current_version,
+            min=av_min, max=av_max)
+        self.reindex(Addon)
+
+        data = self.perform_search(
+            self.url, {'promoted': 'recommended'})
+        assert data['count'] == 2
+        assert len(data['results']) == 2
+        assert {res['id'] for res in data['results']} == {addon.pk, addon2.pk}
+
+        # And with app filtering too
+        data = self.perform_search(
+            self.url, {'promoted': 'recommended', 'app': 'firefox'})
```

Comment:
Should this use constants instead of strings?

Reply:
we tend to just write the strings in the api tests (if they're human understandable strings) so things would loudly fail if an api value was inadvertently changed.